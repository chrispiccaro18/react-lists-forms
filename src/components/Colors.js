import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

function Colors({ colors }) {
  const colorsLis = colors.map(({ name, rgb }) => (
    <li key={`${name}`}>
      <Color name={name} rgb={rgb}/>
    </li>
  ));

  return (
    <ul>
      {colorsLis}
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};

export default Colors;
