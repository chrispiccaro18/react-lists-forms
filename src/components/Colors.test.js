import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

describe('Colors component', () => {
  const colorArr = [
    { name: 'sky blue', rgb: { red: 135, green: 206, blue: 235 } },
    { name: 'sky blue', rgb: { red: 135, green: 206, blue: 235 } },
    { name: 'sky blue', rgb: { red: 135, green: 206, blue: 235 } }
  ];
  
  it('renders a Colors', () => {
    const wrapper = shallow(<Colors colors={colorArr}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
