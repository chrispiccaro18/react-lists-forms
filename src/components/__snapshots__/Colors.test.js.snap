// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component renders a Colors 1`] = `
<ul>
  <li
    key="sky blue"
  >
    <Color
      name="sky blue"
      rgb={
        Object {
          "blue": 235,
          "green": 206,
          "red": 135,
        }
      }
    />
  </li>
  <li
    key="sky blue"
  >
    <Color
      name="sky blue"
      rgb={
        Object {
          "blue": 235,
          "green": 206,
          "red": 135,
        }
      }
    />
  </li>
  <li
    key="sky blue"
  >
    <Color
      name="sky blue"
      rgb={
        Object {
          "blue": 235,
          "green": 206,
          "red": 135,
        }
      }
    />
  </li>
</ul>
`;
